docker/configure_workers_and_start.py

```diff
@@ -389,29 +554,219 @@ def generate_base_homeserver_config() -> None:
     subprocess.run(["/usr/local/bin/python", "/start.py", "migrate_config"], check=True)
 
 
+def parse_worker_types_from_env(
+    requested_worker_types: List[str],
+) -> Dict[str, Dict[str, Any]]:
+    """Read the desired list of worker from environment variables and prepare the data
+    for use in generating worker config files while also checking for potential gotchas.
+
+    Args:
+        requested_worker_types: The string pulled from the environment containing
+            the worker requested data.
+
+    Returns: A  dict of all information needed to generate worker files. Format:
+        {'worker_name':
+            {'worker_base_name': 'base_name'},
+
+            {'worker_types_set':
+                Set{'worker_type', 'other_worker_type'}
+            }
+        }
```

Comment:
I think I may have formatted this wrong. The one things that are fixed are `worker_base_name` and `worker_types_set`, everything else in quotes is a string with appropriate values.

That being said, I'm not against use a Class of some kind to represent this data(I do like type hints). But aren't `attrs` and `dataclass` awfully fancy(and an additional import) for just getting rid of some boilerplate that I would probably override anyways. I would move(almost) all of the error checking into the `__init__`(somethings wouldn't move well, like verifying that a given `worker_base_name` isn't in use by something else).

Reply:
maybe controversial, but I'd rather use a `dataclass` (which is the more lightweight option of the two and part of the stdlib) than a dict with fixed string keys, even at the small potential performance cost. The readability in this file is already suffering and I think this would help.